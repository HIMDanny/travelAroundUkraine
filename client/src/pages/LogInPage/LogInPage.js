import React from 'react';
import { Container, Paper, Typography, Slide } from '@mui/material';
import { grey } from '@mui/material/colors';

import LogInForm from '../../features/LogIn/components/LogInForm';

const paperSX = {
  textAlign: 'center',
  px: { xs: 2 },
  py: { xs: 4 },
  backgroundColor: grey.A100,
  borderRadius: '24px',
  width: { desktop: '70%' },
  m: '0 auto',
};

const LogInPage = () => (
  <Container sx={{ marginTop: { xs: '120px', tablet: '180px' }, height: '60vh' }}>
    <Slide timeout={500} direction="up" in mountOnEnter unmountOnExit>
      <Paper sx={paperSX}>
        <Typography variant="h2">Log In</Typography>
        <LogInForm />
      </Paper>
    </Slide>
  </Container>
);

export default LogInPage;
