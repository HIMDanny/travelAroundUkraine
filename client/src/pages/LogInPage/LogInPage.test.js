import LogInPage from './LogInPage';
import { render } from '@testing-library/react';

jest.mock('@mui/material/Container', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Slide', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Typography', () => ({ children }) => <p>{children}</p>);
jest.mock('../../features/LogIn/components/LogInForm', () => ({ children }) => <div>{children}</div>);

describe('Render LogInPage', () => {
  test('should LogInPage render', () => {
    const { asFragment } = render(<LogInPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
