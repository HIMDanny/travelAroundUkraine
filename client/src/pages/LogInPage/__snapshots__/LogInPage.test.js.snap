// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render LogInPage should LogInPage render 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1jmgxb8-MuiPaper-root"
      >
        <p>
          Log In
        </p>
        <div />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
